test/GoalCohort.spec.js

```diff
@@ -0,0 +1,289 @@
+      );
+      expect(expectedErr.status).to.equal(409);
+    });
+
+    it("creates new team in named cohort", async () => {
+      const user = await User.findById(
+        mongoose.Types.ObjectId("5dd88892c012321c14267155")
+      );
+      await UserCohort.setUserCohort(user, "Study Cohort");
+      const goal = await Goal.findOneByIdOrAlias("5b5a2cd69b1fafcf999d957e");
+      const goalCohort = await GoalCohort.createTeam(user, goal, "Study Team");
+      const cohort = await Cohort.findForName("Study Cohort");
+      expect(goalCohort).to.exist;
+      expect(goalCohort.goal).to.eql(goal._id);
+      expect(goalCohort.cohort).to.eql(cohort._id);
+      expect(goalCohort.membersMax).to.eql(35);
+      expect(goalCohort.memberSlotsRemaining).to.eql(34);
+      expect(goalCohort.members).to.have.length(1);
+      expect(goalCohort.members[0].user).to.eql(user._id);
+      expect(goalCohort.members[0].teamIndex).to.eql(1);
+      expect(goalCohort.teams).to.have.length(2);
+      expect(goalCohort.teams[0].name).to.eql("Minnows");
+      expect(goalCohort.teams[1].name).to.eql("Study Team");
+    });
+
+    it("creates new team in named cohort that has more than 6 teams and 0 slots remaining", async () => {
+      const user = await User.findById(
+        mongoose.Types.ObjectId("5dd88892c012321c14267155")
+      );
+      await UserCohort.setUserCohort(user, "Test Cohort");
+      const cohort = await Cohort.findForName("Test Cohort");
+      const goal = await Goal.findOneByIdOrAlias("5bb6540cbecb4e208da0fb64");
+      let goalCohort = await GoalCohort.createTeam(user, goal, "Test Team");
+      expect(goalCohort).to.exist;
+      expect(goalCohort.goal).to.eql(goal._id);
+      expect(goalCohort.cohort).to.eql(cohort._id);
+      expect(goalCohort.membersMax).to.eql(35);
+      expect(goalCohort.memberSlotsRemaining).to.eql(4);
+      expect(goalCohort.members).to.have.length(1);
+      expect(goalCohort.members[0].user).to.eql(user._id);
+      expect(goalCohort.members[0].teamIndex).to.eql(6);
+      expect(goalCohort.teams).to.have.length(7);
+      expect(goalCohort.teams[6].name).to.eql("Test Team");
+      expect(goalCohort.teams[6].icon).to.eql("LogoTeamRazorfish");
+
+      goalCohort = await GoalCohort.createTeam(user, goal, "Test Team 2");
+      expect(goalCohort).to.exist;
+      expect(goalCohort.goal).to.eql(goal._id);
+      expect(goalCohort.cohort).to.eql(cohort._id);
+      expect(goalCohort.membersMax).to.eql(40);
+      expect(goalCohort.memberSlotsRemaining).to.eql(9);
+      expect(goalCohort.members).to.have.length(1);
+      expect(goalCohort.members[0].user).to.eql(user._id);
+      expect(goalCohort.members[0].teamIndex).to.eql(7);
+      expect(goalCohort.teams).to.have.length(8);
+      expect(goalCohort.teams[7].name).to.eql("Test Team 2");
+      expect(goalCohort.teams[7].icon).to.eql("LogoTeamZephyr");
+    });
+  });
+
+  describe("join team with invite code", function() {
```

Comment:
let's make this `describe("joinTeam", () => `...and then adjust the text in contained `it` specs as needed